### PR TITLE
Prevent excess resource database lines from being generated when nothing happens

### PR DESCRIPTION
**Pull request by nuclearkatie**

### CODE REVIEW

File: src/toolkit/matl_sell_policy.cc [hunk 1]
```diff
@@ -285,7 +285,7 @@ void MatlSellPolicy::GetMatlTrades(
         trade_mat = mat;
       }
 
-      if (ignore_comp_) {
+      if (ignore_comp_ && it->request->target()->comp() != trade_mat->comp()) {
```
Comment (gonuke) [inline src/toolkit/matl_sell_policy.cc:288]: Just to make sure.... do you want to ensure that these are not the same composition pointer? or not the same physical composition?  if the latter, you could use `compmath::AlmostEq()`.  That may be to "fuzzy" though, and maybe you do want to compare exact `Comp` pointers.

**Review by gonuke — COMMENTED** [review]
One clarifying questions, otherwise ready to merge

File: src/toolkit/matl_sell_policy.cc (hunk 1 above)
Comment (nuclearkatie) [reply src/toolkit/matl_sell_policy.cc:288]: Oh! Perfect, I didn't know about `compmath::AlmostEq()`. That's exactly what I'm looking for. I'll use that instead

**Comment by nuclearkatie** [comment]
Oops, using true in [cycamore#618](https://github.com/cyclus/cycamore/pull/618) apparently didn't work, that's raising the issue in the build/test in cycamore. I'll fix that and then this should be good to go

**Review by gonuke — APPROVED** [review]
Thanks @nuclearkatie

**Comment by nuclearkatie** [comment]
@gonuke passing tests now with [cycamore#619](https://github.com/cyclus/cycamore/pull/619) merged

**Comment by nuclearkatie** [comment]
re-running tests

**Comment by bennibbelink** [comment]
@nuclearkatie did this PR introduce an issue downstream?  It looks like downstream tests never passed on this PR (or subsequent PRs - #1782, #1784), all failing the same tests.

**Comment by nuclearkatie** [comment]
Huh, it looks like there are a bunch of `KeyError`s in the python tests. I'll have to think about if this, I don't think anything should have been affected....

**Comment by bennibbelink** [comment]
Not sure if this is helpful... this is one of the failing tests with some print statements.  It fails with `KeyError: 48` at `self.rsrc_qtys[tx['ResourceId']]`.  This is the test using input file `input/physor/2_Sources_3_Reactors.xml`:
```python
def test_rxtr1_xactions(self):
        mox_exp = [0, 1, 1, 1, 0]
        txs = [0, 0, 0, 0, 0]
        resourceIds = [r['ResourceId'] for r in self.resources]
        print(*resourceIds, sep=', ')
        for tx in self.transactions:
            print("tx['Time']", tx['Time'])
            print("tx['ResourceId']", tx['ResourceId'])
            if tx['ReceiverId'] == self.r1 and tx['SenderId'] == self.smox:
                txs[tx['Time']] += self.rsrc_qtys[tx['ResourceId']]
        assert_array_almost_equal(mox_exp, txs)
```

This is the output - 48 is not in the list of resource IDs (and thus not in `self.rsrc_qtys`).  The resources are set via a `SELECT * FROM Resources` query on the output.
```text
2, 4, 46, 47, 50, 51, 54, 55, 58, 59, 62, 63, 66, 67, 70, 71, 74, 75, 78, 79, 82, 83, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 176, 177, 180, 181, 184, 185, 188, 189, 192, 193, 196, 197, 200, 201, 204, 205, 208, 209, 212, 213, 216, 217, 220, 221, 224, 225, 228, 229, 232, 233, 236, 237, 240, 241, 244, 245, 248, 249, 252, 253, 255, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 274, 396, 397, 400, 401, 404, 405, 408, 409, 412, 413, 416, 417, 420, 421, 424, 425, 428, 429, 432, 433, 436, 437, 440, 441, 444, 445, 448, 449, 452, 453, 456, 457, 460, 461, 464, 465, 468, 469, 472, 473, 476, 477, 480, 481, 484, 485, 488, 489, 492, 493, 496, 497, 500, 501, 504, 505, 508, 509, 512, 513, 515, 516, 517, 518, 519, 520, 521, 522, 523, 524, 525, 526, 527, 528, 529, 530, 531, 532, 533, 534, 535, 536, 537, 538, 539, 540, 541, 542, 543, 544, 666, 667, 670, 671, 674, 675, 678, 679, 682, 683, 686, 687, 690, 691, 694, 695, 698, 699, 702, 703, 706, 707, 710, 711, 714, 715, 718, 719, 722, 723, 726, 727, 730, 731, 734, 735, 738, 739, 742, 743, 746, 747, 750, 751, 754, 755, 758, 759, 762, 763, 766, 767, 770, 771, 774, 775, 778, 779, 782, 783
tx['Time'] 1
tx['ResourceId'] 48
```

**Comment by nuclearkatie** [comment]
@bennibbelink did you figure anything else out about the failing tests?

**Comment by bennibbelink** [comment]
No, I've been preoccupied with #1791 recently so I haven't had a chance to dive deep into this again.  I am not 100% certain where these problems originated though (still leaning towards this PR or #1784), so any ideas would be appreciated.  I'm also not sure if its a problem in cyclus or cycamore (since its only downstream tests that fail), did you make corresponding changes in cycamore for all of your recent packaging updates?

**Pull request merged**